mm2src/coins/eth/nft/nft_structs.rs

```diff
@@ -0,0 +1,220 @@
+use crate::{TxFeeDetails, WithdrawFee};
+use mm2_number::BigDecimal;
+use rpc::v1::types::Bytes as BytesJson;
+use serde::Deserialize;
+use std::str::FromStr;
+
+#[derive(Debug, Deserialize)]
+pub struct NftListReq {
+    pub(crate) chains: Vec<Chain>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct NftMetadataReq {
+    pub(crate) token_address: String,
+    pub(crate) token_id: BigDecimal,
+    pub(crate) chain: Chain,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub enum Chain {
+    Bnb,
+    Eth,
+}
+
+#[derive(Debug, Display)]
+pub enum ParseContractTypeError {
+    UnsupportedContractType,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub enum ContractType {
+    Erc1155,
+    Erc721,
+}
+
+impl FromStr for ContractType {
+    type Err = ParseContractTypeError;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<ContractType, ParseContractTypeError> {
+        match s {
+            "ERC1155" => Ok(ContractType::Erc1155),
+            "ERC721" => Ok(ContractType::Erc721),
+            _ => Err(ParseContractTypeError::UnsupportedContractType),
+        }
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct Nft {
+    pub(crate) chain: Chain,
+    pub(crate) token_address: String,
+    pub(crate) token_id: BigDecimal,
+    pub(crate) amount: BigDecimal,
+    pub(crate) owner_of: String,
+    pub(crate) token_hash: String,
+    pub(crate) block_number_minted: u64,
+    pub(crate) block_number: u64,
+    pub(crate) contract_type: Option<ContractType>,
+    pub(crate) name: Option<String>,
+    pub(crate) symbol: Option<String>,
+    pub(crate) token_uri: Option<String>,
+    pub(crate) metadata: Option<String>,
+    pub(crate) last_token_uri_sync: Option<String>,
+    pub(crate) last_metadata_sync: Option<String>,
+    pub(crate) minter_address: Option<String>,
+}
+
+/// This structure is for deserializing NFT json to struct.
+/// Its needed to convert fields properly, because all fields in json have string type.
+#[derive(Debug, Deserialize)]
+pub struct NftWrapper {
+    pub(crate) token_address: String,
+    pub(crate) token_id: Wrap<BigDecimal>,
+    pub(crate) amount: Wrap<BigDecimal>,
+    pub(crate) owner_of: String,
+    pub(crate) token_hash: String,
+    pub(crate) block_number_minted: Wrap<u64>,
+    pub(crate) block_number: Wrap<u64>,
+    pub(crate) contract_type: Option<Wrap<ContractType>>,
+    pub(crate) name: Option<String>,
+    pub(crate) symbol: Option<String>,
+    pub(crate) token_uri: Option<String>,
+    pub(crate) metadata: Option<String>,
+    pub(crate) last_token_uri_sync: Option<String>,
+    pub(crate) last_metadata_sync: Option<String>,
+    pub(crate) minter_address: Option<String>,
+}
+
+#[derive(Debug)]
+pub struct Wrap<T>(pub(crate) T);
```

Comment:
This is used in `coins` crate only. This could be `pub(crate)`.
Also I think it could be renamed to `SerdeStringWrap`.

Reply:
Done